Compile error with OpenSSL 4.0

It seems current release isn't compatible with OpenSSL 4.0.
Switching to OpenSSL 3.x fixed this error.

```
/builddir/build/BUILD/monero-0.18.5.1-build/monero-source-v0.18.5.1/contrib/epee/src/net_ssl.cpp: In function ‘bool epee::net_utils::create_rsa_ssl_certificate(EVP_PKEY*&, X509*&)’:
/builddir/build/BUILD/monero-0.18.5.1-build/monero-source-v0.18.5.1/contrib/epee/src/net_ssl.cpp:175:42: error: invalid conversion from ‘const X509_NAME*’ {aka ‘const X509_name_st*’} to ‘X509_NAME*’ {aka ‘X509_name_st*’} [-fpermissive]
  175 |   X509_NAME *name = X509_get_subject_name(cert);
      |                     ~~~~~~~~~~~~~~~~~~~~~^~~~~~
      |                                          |
      |                                          const X509_NAME* {aka const X509_name_st*}
```

Full log:
`https://download.copr.fedorainfracloud.org/results/zpc00/monero/fedora-rawhide-x86_64/10705837-monero/builder-live.log.gz`

--- a/contrib/epee/src/net_ssl.cpp
+++ b/contrib/epee/src/net_ssl.cpp
@@ -174,8 +174,7 @@ bool create_rsa_ssl_certificate(EVP_PKEY *&pkey, X509 *&cert)
     X509_free(cert);
     return false;
   }
-  X509_NAME *name = X509_get_subject_name(cert);
-  X509_set_issuer_name(cert, name);
+  X509_set_issuer_name(cert, X509_get_subject_name(cert));
 
   if (X509_sign(cert, pkey, EVP_sha256()) == 0)
   {

